src/utils/coingecko_types.ts

```diff
@@ -0,0 +1,109 @@
+import type { FiatID, TokenID } from './fiat_oracle_types';
+
+export interface CoinGeckoResponse {
+	/**
+	 * Whether the response status code returned a successful code (>200 && <300).
+	 */
+	success: boolean;
+	/**
+	 * The response status message
+	 */
+	message: string;
+	/**
+	 * The response status code
+	 */
+	code: number;
+	/**
+	 * The body data in JSON format from the request.
+	 */
+	data: any;
+}
+
+export interface CoinGeckoPriceRequestParams extends Record<string, string> {
+	// CSV of the coin IDs
+	ids: string;
+	// CSV of the fiat currencies to be quoted in
+	vs_currencies: string;
+}
+
+export type CoinGeckoPriceResponseData = {
+	[token in TokenID]: CoinGeckoCoinValue;
+};
+
+type CoinGeckoCoinValue = {
+	[currency in FiatID]: number;
+};
+
+/* The below values were retrived from the CoinGecko API on Friday 13 of August.
```

Comment:
Year might be helpful for posterity ;)

Reply:
There: 01580efd25e72604403eb75fabc9f8e19a46efb4